pyreaclib outputs Fortran 90 using sundials cvode to integrate the reaction network

dwillcox opened the pull request:
Uses sundials v. 2.6.2 in stiff BDF mode with a RHS and jacobian supplied by reaclib.py. Fortran code relies on template files stored in templates/sundials-cvode.

Demonstration:
examples/CNO_f90
examples/C_f90
examples/URCA_f90

Instructions for making a Fortran network are in examples/README.md

The reaclib.py interface for Fortran simply requires creating a RateCollection of Reaclib rate files (and optionally Reaclib-like table files) and then using the RateCollection.make_network_f90() function.

zingale commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
 *.pyc
+*.o
+*.mod
+*.exe
```
add a new line here

dwillcox replied:
Done!

dwillcox replied:
It occurs to me you meant to add a new line after the *.pyc ignore. I've done so.

zingale commented:
we shouldn't include PDF files -- those don't diff/merge well

we should clean up the makefile -- it shoud not need to have separate rules for everything.  We can do the dependency checking automatically.  Look at dep.py and the makefile in ode-openacc to see how to hook in the automagic dep stuff

we also shouldn't duplicate all of the reaclib files for the network -- can we point this to the versions already in the repo (in CNO/) or perhaps centralize them?

dwillcox commented:
PDFs are gone, I'm working on centralizing the reaclib rate files in pyreaclib/reaclib-rates and having the Rate class read first from that directory and next from the working directory if unsuccessful. 

Next I'll get the Makefile automagicked.

dwillcox commented:
Alright, I've implemented the automake system and changed the behavior of reaclib.py to streamline creating a new Fortran network and eliminate issues with additional Fortran programs I wrote for debugging. reaclib.py now copies the needed files into the working directory. Programs useful for debugging are still in templates/sundials-cvode but aren't copied to the working directory because they aren't for general rate collections as of yet.

I edited the make files and dep.py to make it easier to compile when multiple program source files exist in the working directory and to specify where the SUNDIALS, LAPACK, BLAS libraries are.

zingale commented:
ok, one more nit -- some of what you are committing looks like it is the output of the script, e.g. the automagically generated .F90.  Do we want the output in the repo? I didn't include the output in the pure python example.

However, at some point in the future, we should introduce unit testing to compare things.

dwillcox commented:
I've removed the script outputs from the examples directories and cleaned up the URCA readme so (with the removal of net.par) the default runtime values of net.par from the template won't cause the program to throw SIGFPE if the user doesn't add an abundance stopping condition for the root solver.

dwillcox commented:
Unit testing would be useful, I'll think about how to do it along the lines of the MAESTRO test_react unit test.
